src/selection.coffee

```diff
@@ -387,7 +387,7 @@ class Selection extends Model
           @editor.setIndentationForBufferRow(row, newIndent)
     else if options.autoIndentNewline and text == '\n'
```

Comment:
If you could avoid this section running entirely, wouldn't that mean you didn't need to change `language-mode.coffee` at all?